### ISSUE TITLE
Cell execution counter does not reset when pasting cells

### ISSUE THREAD

**Issue by aiqc** [jupyterlab/jupyterlab#13149]
## Description

![image](https://user-images.githubusercontent.com/74990642/193038610-f60b2496-7aaa-4937-98b1-dcfc18a897db.png)

^ The selected cells were pasted and their `[102]` execution id does not reset. 
^ The expected behavior would be resetting to `[]` since they have not been executed . 

---

## Reproduce

1. Run some cells
2. Copy + paste them
3. Observe that their execution counter is identical

---

### Env

- MacOS 11.6.5 (20G527)
- Firefox 105.0 (64-bit) + Chrome 105.0.5195.125 (Official Build) (x86_64)
- JupyterLab 3.4.2

**Comment by jweill-aws** [comment]
The cell indicator also maps to the `Input` and `Output` special arrays in Python; should pasting in cells modify the contents of these arrays immediately or only when the new cells are executed?

**Comment by krassowski** [comment]
> ^ The expected behavior would be resetting to [] since they have not been executed .

I don't think this is accurate expectation. When you copy-paste cells which were executed you also retain outputs. When cell contents change after execution we do not reset the counter but add an asterisk with orange highlight to indicate dirty state. I think this is what we should do here too.

**Comment by aiqc** [comment]
@krassowski good idea on indicating its state somehow. The `*` is 100% synonymous with running in my mind. If I saw `*` sparsely throughout a notebook, then I would think something is seriously broken. If a symbol is to be used, I'd recommend a different one like `~`.

**Comment by krassowski** [comment]
No, this is not what I meant. I meant this one:

![dot](https://user-images.githubusercontent.com/21197331/119834056-e9992d80-beff-11eb-8e28-3d3d0c51722d.gif)

from https://github.com/jupyterlab/jupyterlab/pull/10296. My logic is that the paste is likely invalidating output.

I should have said dot not asterisks, sorry.

**Comment by krassowski** [comment]
> ^ The same behavior is exhibited when pasting cells into a separate notebook


This is tracked in https://github.com/jupyterlab/jupyterlab/issues/11570
